pkg/provider/branch/trunk/trunk.go

```diff
@@ -479,6 +481,31 @@ func (t *trunkENI) CreateAndAssociateBranchENIs(pod *v1.Pod, securityGroups []st
 	return newENIs, nil
 }
 
+// DeleteAllBranchENIs deletes all the branch ENIs associated with the trunk and all the ENIs present in the cool down
+// queue, this is the last API call to the the Trunk ENI before it is removed from cache
+func (t *trunkENI) DeleteAllBranchENIs() {
```

Comment:
I'm good with this change itself.

However, this we just logs in terms of failure, in extreme edge cases, we can still leak eni right?

Reply:
yes. By default all ec2 calls do retry 3 times, if error is retryable like throttling,etc. But yes we can fail to clean and leak. https://github.com/aws/amazon-vpc-resource-controller-k8s/blob/a08a9de907f3812be3ae894cc00c93fb64e4a027/pkg/aws/ec2/api/helper.go#L36